datastore/longhorn.go

```diff
@@ -6449,6 +6449,31 @@ func (s *DataStore) IsV2DataEngineDisabledForNode(nodeName string) (bool, error)
 	return false, nil
 }
 
+func (s *DataStore) IsVolumeCompatibleWithNodeEngine(volumeName string, nodeName string) (bool, error) {
+	// v1 volumes are always considered compatible.
+	// v2 volumes are compatible only if the node has NOT disabled v2 engine.
+
+	volume, err := s.GetVolumeRO(volumeName)
+	if err != nil {
+		// Because the volume doesn't exist,
+		// we can skip the following node check for backupvolumes and backups synced from a remote backup target.
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}
+
+	if types.IsDataEngineV2(volume.Spec.DataEngine) {
+		v2Disabled, err := s.IsV2DataEngineDisabledForNode(nodeName)
+		if err != nil {
+			return false, errors.Wrapf(err, "failed to check v2 data engine disable label for node %s", nodeName)
+		}
+		return !v2Disabled, nil
+	}
+
+	return true, nil
+}
```

Comment:
no need to get volume for v1 data engine
```suggestion
func (s *DataStore) IsVolumeCompatibleWithNodeEngine(volumeName string, nodeName string) (bool, error) {
	// v1 volumes are always considered compatible.
	if types.IsDataEngineV1(volume.Spec.DataEngine) {
		return true, nil
	}
	
	
	// v2 volumes are compatible only if the node has NOT disabled v2 engine.
	volume, err := s.GetVolumeRO(volumeName)
	if err != nil {
		// Because the volume doesn't exist,
		// we can skip the following node check for backupvolumes and backups synced from a remote backup target.
		if apierrors.IsNotFound(err) {
			return true, nil
		}
		return false, err
	}

	
		v2Disabled, err := s.IsV2DataEngineDisabledForNode(nodeName)
		if err != nil {
			return false, errors.Wrapf(err, "failed to check v2 data engine disable label for node %s", nodeName)
		}
		return !v2Disabled, nil
}
```

Reply:
If not get the volume, how do I know the `dataEngine`?